bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -134,14 +134,17 @@ export class DataTableView extends WidgetView {
 
     // This is obnoxious but there is no better way to programmatically force
     // a re-sort on the existing sorted columns until/if we start using DataView
-    const columns = this.grid.getColumns()
-    const sorters = this.grid.getSortColumns().map((x: any) => ({
-      sortCol: {
-        field: columns[this.grid.getColumnIndex(x.columnId)].field,
-      },
-      sortAsc: x.sortAsc,
-    }))
-    this.data.sort(sorters)
+    if (this.model.sortable) {
+      const columns = this.grid.getColumns()
+      const sorters = this.grid.getSortColumns().map((x: any) => ({
```

Comment:
This `any` type can be safely removed, because `getSortColumns()` has a well defined type declaration.